### PR TITLE
Port update_submodules.sh to update_submodules.cmd.

### PR DESCRIPTION
**Pull request by jaykrell**
It actually doesn't look quite right, but does mimic update_submodules.sh.

### CODE REVIEW

File: scripts/update_submodules.cmd [hunk 1]
```diff
@@ -0,0 +1,17 @@
+@if not exist .git goto :eof
+
+git submodule update --init --recursive && @goto :ok
+git submodule init && git submodule update --recursive && @goto :ok
+
+:: This looks wrong but is what update_submodules.sh does.
+git submodule init && git submodule update && @goto :error
```
Comment (akoeplinger) [inline scripts/update_submodules.cmd]: it's a best effort thing which is just there for very old git versions which supported neither `git submodule update --init --recursive` nor `... init  && ... update --recursive`.

I think we can remove it.

Comment (jaykrell) [reply scripts/update_submodules.cmd]: I thought it was just there for developer convenience.

Comment (jaykrell) [reply scripts/update_submodules.cmd]: On the first comment: I thought it was just there for developer convenience.

Comment (akoeplinger) [reply scripts/update_submodules.cmd]: Nah it's getting run by the root Makefile at the start of the build: https://github.com/mono/mono/blob/feaf8ee391a09fbff2137905ac7139f9bacc99fa/Makefile.am#L21

That's why you probably never needed to manually pull in the submodule changes ;)

**Pull request closed without merging**